test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -237,25 +449,114 @@ contract LoansForkTest is LoansTestBase {
         // Verify balance changes
         assertEq(userCashBefore - pair.cashAsset.balanceOf(user), loanAmount);
         assertEq(pair.underlying.balanceOf(user) - userUnderlyingBefore, underlyingOut);
     }
 
+    function createEscrowOffer(uint duration) internal returns (uint offerId) {
+        vm.startPrank(escrowSupplier);
+        pair.underlying.approve(address(pair.escrowNFT), underlyingAmount);
+        offerId = pair.escrowNFT.createOffer(
+            underlyingAmount,
+            duration,
+            interestAPR,
+            maxGracePeriod,
+            lateFeeAPR,
+            0 // minEscrow
+        );
+        vm.stopPrank();
+    }
+
+    function openEscrowLoan(uint minLoanAmount, uint providerOfferId, uint escrowOfferId, uint escrowFee)
+        internal
+        returns (uint loanId, uint providerId, uint loanAmount)
+    {
+        vm.startPrank(user);
+        // Approve underlying amount plus escrow fee
+        pair.underlying.approve(address(pair.loansContract), underlyingAmount + escrowFee);
+
+        (loanId, providerId, loanAmount) = pair.loansContract.openEscrowLoan(
+            underlyingAmount,
+            minLoanAmount,
+            ILoansNFT.SwapParams(0, address(pair.loansContract.defaultSwapper()), ""),
+            ILoansNFT.ProviderOffer(pair.providerNFT, providerOfferId),
+            ILoansNFT.EscrowOffer(pair.escrowNFT, escrowOfferId),
+            escrowFee
+        );
+        vm.stopPrank();
+    }
+
+    function _createEscrowOffers() internal returns (uint offerId, uint escrowOfferId) {
+        uint providerBalanceBefore = pair.cashAsset.balanceOf(provider);
+        offerId = createProviderOffer(pair, callstrikeToUse, offerAmount);
+        assertEq(pair.cashAsset.balanceOf(provider), providerBalanceBefore - offerAmount);
+
+        // Create escrow offer
+        escrowOfferId = createEscrowOffer(pair.durations[0]);
+    }
+
+    function _executeEscrowLoan(uint offerId, uint escrowOfferId)
+        internal
+        returns (uint loanId, uint providerId, uint loanAmount)
+    {
+        uint expectedEscrowFee = pair.escrowNFT.interestFee(escrowOfferId, underlyingAmount);
+
+        // Open escrow loan using base function
+        (loanId, providerId, loanAmount) = openEscrowLoan(
+            0.3e6, // minLoanAmount
+            offerId,
+            escrowOfferId,
+            expectedEscrowFee
+        );
+    }
+
+    function _verifyEscrowLoan(
+        uint loanId,
+        uint providerId,
+        uint loanAmount,
+        uint escrowOfferId,
+        uint feeRecipientBalanceBefore,
+        uint expectedProtocolFee
+    ) internal view {
+        uint expectedEscrowFee = pair.escrowNFT.interestFee(escrowOfferId, underlyingAmount);
+
+        uint userUnderlyingBefore = pair.underlying.balanceOf(user) + underlyingAmount + expectedEscrowFee;
+        uint escrowSupplierUnderlyingBefore = pair.underlying.balanceOf(escrowSupplier);
+
+        assertGt(loanAmount, 0);
```

Comment:
might be good to calculate this, even if not exactly, to be within some tighter bounds. For both escrow and non escrow loans